src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseStringContainsCharOverloadWithSingleCharactersAnalyzer.cs

```diff
@@ -0,0 +1,97 @@
+        private static readonly LocalizableString s_localizableMessage_CA1841 = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.UseStringContainsCharOverloadWithSingleCharactersMessage), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription_CA1841 = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.UseStringContainsCharOverloadWithSingleCharactersDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+
+        internal static readonly DiagnosticDescriptor s_rule_CA1841 = DiagnosticDescriptorHelper.Create(
+            CA1841,
+            s_localizableTitle_CA1841,
+            s_localizableMessage_CA1841,
+            DiagnosticCategory.Performance,
+            RuleLevel.IdeSuggestion,
+            description: s_localizableDescription_CA1841,
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(s_rule_CA1841);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+            context.RegisterCompilationStartAction(CheckIfRuleIsApplicableAndRegister);
+        }
+
+        private void CheckIfRuleIsApplicableAndRegister(CompilationStartAnalysisContext context)
+        {
+            var stringType = context.Compilation.GetSpecialType(SpecialType.System_String);
+            var charType = context.Compilation.GetSpecialType(SpecialType.System_Char);
+
+            // Bail out of further evaluations in special cases where string/char types arent even available
+            if (stringType is null || charType is null)
+                return;
+
+            var ruleIsApplicable = stringType.GetMembers("Contains").OfType<IMethodSymbol>().Any(m =>
+            {
+                if (m.Parameters.Length > 0)
+                {
+                    return m.Parameters[0].Type.SpecialType == SpecialType.System_Char;
+                }
+                return false;
+            });
+
+            if (ruleIsApplicable)
+                context.RegisterOperationAction(AnalyseContainsUsage, OperationKind.Invocation);
+        }
+
+        private void AnalyseContainsUsage(OperationAnalysisContext context)
+        {
+            var invocationOperation = (IInvocationOperation)context.Operation;
+            if (ShouldAnalyzeInvocations(invocationOperation) && CheckForViolationAndTransformLiteral(invocationOperation.Arguments[0], out var sourceString))
```

Comment:
`invocationOperation.Arguments[0]` for below invocation to `M` will give you first named argument that is passed to the second parameter `p2`:

```csharp
void M(char p1, int p2);
M(p2: 1, 'c');

Reply:
I think you need to iterate through `invocationOperation.Arguments` and find the argument whose `.Parameter` has index `0` OR parameter symbol matches.

Reply:
Actually, you can just use this extension method that @buyaa-n recently added: https://github.com/dotnet/roslyn-analyzers/blob/04883a7d9343d5e757f2c1a5401b74d9e8d51eaa/src/Utilities/Compiler/Extensions/IOperationExtensions.cs#L787-L789

Reply:
Please do add a unit test for this case.

Reply:
oh hell yeah that could've went south real fast :D

Reply:
Is it legal for me to use both CSharp and VB Syntax Types in a language agnostic analyzer or do i have to split this off into separate analyzers? Because theres no IOperation way to detect named parameters and i need to detect them to be able to tell the fixer that it needs to generate a named argument.

Reply:
> use both CSharp and VB Syntax Types in a language agnostic analyzer

No, please don't do so. Instead make the language agnostic type abstract and add non-abstract deriving types in C# and VB projects in this layer for language specific functionality.

Reply:
Okay - will do so after work today.